src/item.cpp

```diff
@@ -2535,6 +2535,26 @@ void item::magazine_info( std::vector<iteminfo> &info, const iteminfo_query *par
         return;
     }
 
+    if( parts->test( iteminfo_parts::MAGAZINE_COMPATIBLE_GUNS ) ) {
+        const auto compare = []( const auto & lhs, const auto & rhs ) {
+            return localized_compare( lhs, rhs );
+        };
+        std::set<std::string, decltype( compare )> compatible_guns( compare );
+        for( const itype *it : item_controller->all() ) {
```

Comment:
This seems really slow and should be cached IMO

Reply:
#59743 if you want some inspiration for caching.

Reply:
Stole the idea to do it as part of loading, thanks